hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityConverter.java

```diff
@@ -74,15 +81,30 @@
                 new ConvertedEntityResultSet(entities, response.getTotal(), entities.size()));
   }
 
+  private Optional<BaselineEntity> getBaselineEntity(Map<String, BaselineEntity> baselineEntityMap,
+                                           org.hypertrace.gateway.service.v1.entity.Entity entity) {
+    if (!baselineEntityMap.isEmpty() && baselineEntityMap.containsKey(entity.getId())) {
+      return Optional.of(baselineEntityMap.get(entity.getId()));
```

Comment:
nit `Optional.ofNullable` would replace the if here since a get for a missing id would just return null.